pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -275,7 +275,10 @@ func (r *ReconcileWindowsMachineConfig) addWorkerNode() (types.WindowsVM, error)
 	}
 
 	log.V(1).Info("configuring the Windows VM", "ID", vm.GetCredentials().GetInstanceId())
-	nc := nodeconfig.NewNodeConfig(r.k8sclientset, vm)
+	nc, err := nodeconfig.NewNodeConfig(r.k8sclientset, vm)
+	if err != nil {
+		log.Error(err, "failed to create node config", "VM", vm.GetCredentials().GetInstanceId())
```

Comment:
This needs to be handled or it will cause a NPE

Reply:
I think you misunderstood,  the error needs to be handled, we need to mark as a reconcile error and clean up the VM